waku/v2/protocol/waku_rln_relay/waku_rln_relay_utils.nim

```diff
@@ -934,6 +919,32 @@ proc subscribeToGroupEvents(ethClientUri: string, ethAccountAddress: Address, co
   do (err: CatchableError):
     error "Error from subscription: ", err=err.msg
 
+proc subscribeToGroupEvents(ethClientUri: string, ethAccountAddress: Address, contractAddress: Address, blockNumber: string = "0x0", handler: RegistrationEventHandler) {.async, gcsafe.} = 
+  ## connects to the eth client whose URI is supplied as `ethClientUri`
+  ## subscribes to the `MemberRegistered` event emitted from the `MembershipContract` which is available on the supplied `contractAddress`
+  ## it collects all the events starting from the given `blockNumber`
+  ## for every received event, it calls the `handler`
+  let web3 = await newWeb3(ethClientUri)
+  var latestBlock: Quantity
+  let newHeadCallback = proc (blockheader: BlockHeader) {.gcsafe.} =
+    latestBlock = blockheader.number
+    debug "block received", blockNumber = latestBlock
+  let newHeadErrorHandler = proc (err: CatchableError) {.gcsafe.} =
+    error "Error from subscription: ", err=err.msg
+  discard await web3.subscribeForBlockHeaders(newHeadCallback, newHeadErrorHandler)
```

Comment:
This subscription is handled separately from the subscription to the `MemberRegistered` events on the contract. The `latestBlock` perceived by `subscribeForBlockHeaders` is not necessarily the same as the same block processed in subscription to the `MemberRegistered` events (they are two separate queries and streams of data with no correlation). Have you checked that?

Reply:
Yes, I mentioned this in our call. When we actually process blocks atomically, I will add another field to the WakuRlnRelay object called `lastProcessedBlock`. `lastSeenBlock` will always be an indicator of the health of the websocket connection.